src/Compilers/CSharp/Portable/FlowAnalysis/VariablesDeclaredWalker.cs

```diff
@@ -77,7 +77,12 @@ private void NoteDeclaredPatternVariables(BoundPattern pattern)
             if (IsInside && pattern.Kind == BoundKind.DeclarationPattern)
             {
                 var decl = (BoundDeclarationPattern)pattern;
-                _variablesDeclared.Add(decl.LocalSymbol);
+                if (decl.Variable.Kind == SymbolKind.Local)
```

Comment:
I don't recall modifying this file/walker for d-declaration in script. Should I have?